contrib/nydus-backend-proxy/src/main.rs

```diff
@@ -0,0 +1,240 @@
+#[macro_use]
+extern crate rocket;
+extern crate http_range;
+extern crate lazy_static;
+extern crate nix;
+
+use rocket::fs::FileServer;
+use std::env;
+
+mod getblob {
+    use http_range::HttpRange;
+    use lazy_static::lazy_static;
+    use nix::sys::uio;
+    use rocket::fs::NamedFile;
+    use rocket::http::Status;
+    use rocket::request::{self, FromRequest, Outcome};
+    use rocket::response::{self, stream::ReaderStream, Responder};
+    use rocket::Request;
+    use rocket::Response;
+    use std::collections::HashMap;
+    use std::os::unix::io::AsRawFd;
+    use std::path::{Path, PathBuf};
+    use std::{env, ffi, fs, io};
+
+    lazy_static! {
+        static ref BLOBFDMAP: HashMap<ffi::OsString, fs::File> = set_blobmap();
+        static ref IMAGEPATH: PathBuf = set_imagepath();
+    }
+
+    fn set_imagepath() -> PathBuf {
+        let args: Vec<String> = env::args().collect();
+        let path = Path::new(&args[1]).join("blobs").to_path_buf();
```

Comment:
This path should be got from the main func, can we define a struct and use it in routes?

``` rust
struct BlobBackend {
    root: PathBuf,
    blobs: HashMap<ffi::OsString, fs::File>,
}

impl BlobBackend {
    fn new(root: PathBuf) -> Self {
        Self {
            root,
        }
    }

    fn get(...) ...

    fn check(...) ...
}
```